cmd/env.go

```diff
@@ -0,0 +1,49 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/celestiaorg/celestia-node/node"
+)
+
+type Env struct {
+	ndType    node.Type
+	storePath string
+
+	opts []node.Option
+}
+
+func WithEnv(ctx context.Context) context.Context {
+	_, err := GetEnv(ctx)
+	if err == nil {
+		panic("cmd: only one Env is allowed to be set in a ctx")
+	}
+
+	return context.WithValue(ctx, env, &Env{})
+}
+
+func GetEnv(ctx context.Context) (*Env, error) {
+	env, ok := ctx.Value(env).(*Env)
+	if !ok {
+		return nil, fmt.Errorf("cmd: Env is not set in ctx.Context")
+	}
+
+	return env, nil
+}
+
+func (env *Env) SetNodeType(tp node.Type) {
+	env.ndType = tp
+}
+
+func (env *Env) Options() []node.Option {
+	return env.opts
+}
+
+func (env *Env) addOption(opt node.Option) {
+	env.opts = append(env.opts, opt)
+}
+
+type envKey struct{}
+
+var env = envKey{}
```

Comment:
what's the point of this?

Reply:
To preallocate env, so when `WithEnv` is called it does not allocate.

Reply:
Isn't this an optimization with negligible gain but which adds cognitive overhead by making it more difficult to reason about this? Especially as `env` is also used a local var name everywhere as well.

Reply:
We can have another name. Agree that having it named as `env` may add additional cognitive overhead and be confusing. If the name is different, then it should be ok

Reply:
Changed